assets/hip-206/solidity/TokenCreateContract.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.6.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "./HederaTokenService.sol";
+import "./IHederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+
+contract TokenCreateContract is HederaTokenService {
+
+    using Bits for uint;
+
+    // Create Fungible Token with no custom fees, with a user account as admin, contract as supply and pause key.
+    function createFungible(address contractKey) external returns (address createdTokenAddress) {
+
+        // create TokenKey of type adminKey with value inherited from called account
+        uint adminKeyType;
+        IHederaTokenService.KeyValue memory adminKeyValue;
+        // turn on bit corresponding to admin key type
+        adminKeyType = adminKeyType.setBit(0);
+        // set the value of the key to be inherited from the calling account
+        adminKeyValue.inheritAccountKey = true;
+        
+        // create TokenKey of types supplyKey and pauseKey with value a contract address passed as function arg
+        uint supplyPauseKeyType;
+        IHederaTokenService.KeyValue memory supplyPauseKeyValue;
+        // turn on bits corresponding to supply and pause key types
+        supplyPauseKeyType = supplyPauseKeyType.setBit(4);
+        supplyPauseKeyType = supplyPauseKeyType.setBit(6);
+        // set the value of the key to the contract address passed as function arg
+        supplyPauseKeyValue.contractId = contractKey;
+
+        // instantiate the list of keys we'll use for token create
+        IHederaTokenService.TokenKey[] memory keys;
+        keys[0] = IHederaTokenService.TokenKey (adminKeyType, adminKeyValue);
+        keys[1] = IHederaTokenService.TokenKey (supplyPauseKeyType, supplyPauseKeyValue); 
+
+        // declare fee fields
+        IHederaTokenService.FixedFee[] memory fixedFees;
+        IHederaTokenService.FractionalFee[] memory fractionalFees;
+
+        IHederaTokenService.HederaToken memory myToken;
+        myToken.name = "MyToken";
+        myToken.symbol = "MTK";
+        myToken.treasury = address(this);
+        myToken.tokenKeys = keys;
+
+        (bool success, address token, bytes memory errorMessage) = 
+            HederaTokenService.createFungibleToken(myToken, 200, 8, fixedFees, fractionalFees);
```

Comment:
I think we should have a varient without custom fees, or fold it into the token description.